bluesky/callbacks/best_effort.py

```diff
@@ -441,6 +441,55 @@ def clear(self):
         self._buffer = StringIO()
         self._baseline_toggle = True
 
+    def plot_prune_fifo(self, num_lines, y_signal, x_signal):
+        """
+        Find the plot with axes x_signal and y_signal.  Replot with only the last *num_lines* lines.
+
+        .. note:: This is not a bluesky plan.  Call it as a normal Python function.
+
+        Example to remove all scans but the last:
+        >>> bec.plot_prune_fifo(1, noisy, m1)
+
+        Parameters
+        ----------
+        num_lines: int
+            number of lines (plotted scans) to keep
+        y_signal: object
+            instance of ophyd.Signal (or subclass), 
+            dependent (y) axis
+        x_signal: object
+            instance of ophyd.Signal (or subclass), 
+            independent (x) axis
+
+        Returns
+        -------
+        lp: object or None
+            instance of LivePlot or None (if not found)
+
+        """
+        assert num_lines >= 0, "num_lines must be 0 or greater"
+        for liveplot in self._live_plots.values():
+            lp = liveplot.get(y_signal.name)
+            if lp is None or lp.x != x_signal.name or lp.y != y_signal.name:
+                continue
+
+            # pick out only the lines that contain plot data
+            # skipping the lines that show peak centers
+            lines = [
+                tr
+                for tr in lp.ax.lines
+                if len(tr._x) != 2 
+                    or len(tr._y) != 2 
+                    or (len(tr._x) == 2 
+                        and tr._x[0] != tr._x[1])
+            ]
+            if len(lines) > n:
+                print(f"limiting LivePlot({y_signal.name}) to {num_lines} lines")
+                lp.ax.lines = lines[-num_lines:]
```

Comment:
Please use the remove method on the `Line2D` object rather than directly touching the `ax.lines` list.

Reply:
Sounds good.

Reply:
Like this?

```
                keepers = lines[-num_lines:]
                for tr in lp.ax.lines:
                    if tr not in keepers:
                        tr.remove()
```

If plot centers are drawn, they will be recreated with the next scan.